lib/pbench/agent/fs.py

```diff
@@ -21,3 +15,26 @@ def safe_mkdir(directory, clean=False):
     except OSError as e:
         if e.errno != errno.EEXIST:
             raise
+
+
+def safe_delete(filename):
+    """Delete a file safely"""
+    try:
+        os.unlink(filename)
+        return 0
+    except OSError as e:
+        if e.errno != errno.ENOENT:
+            return 1
+        return 1
+
+
+def safe_rmtree(directory):
+    """Delete a directory, if it's present otherwise no-op"""
```

Comment:
Sounds like the `else` should be returning `0` if 'not existing' is a no-op???

In which case you could get rid of the first `return 0` and the `else` and simply return 1 on an `Exception` otherwise return `0`...